src/java.base/share/classes/com/sun/crypto/provider/AESCrypt.java

```diff
@@ -1365,8 +1365,15 @@ private void makeSessionKey(byte[] k) throws InvalidKeyException {
         int ROUND_KEY_COUNT = (ROUNDS + 1) * 4;
 
         int BC = 4;
-        int[][] Ke = new int[ROUNDS + 1][4]; // encryption round keys
-        int[][] Kd = new int[ROUNDS + 1][4]; // decryption round keys
+        int[][] Ke = new int[ROUNDS + 1][]; // encryption round keys
+        int[][] Kd = new int[ROUNDS + 1][]; // decryption round keys
+
+        // It is significantly faster to allocate individual arrays,
+        // instead of doing the multi-array allocation. See JDK-8308105.
```

Comment:
Alternatively, could the multi-array allocation get improved?

Reply:
Yes, it could, but it would take significantly more work: there are multiple issues of differing complexity, impact, and risk. This workaround, on the other hand, is straight-forward and backportable.

Reply:
Please look again, I think we can improve this code even further, and that is where JVM optos would not help us much :)